spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/aggregation/ExposedFieldsAggregationOperationContext.java

```diff
@@ -96,6 +96,25 @@ private FieldReference getReference(@Nullable Field field, String name) {
 			return exposedField;
 		}
 
+		if(rootContext instanceof RelaxedTypeBasedAggregationOperationContext) {
```

Comment:
How about `rootContext.getReference(…)`? Wouldn't that work as well?

Reply:
IIRC it broke one of the existing tests for `_id` field projection.